crates/services/src/attestation/mod.rs

```diff
@@ -208,101 +224,92 @@ impl ports::AttestationServiceTrait for AttestationService {
-            signing_address,
-            signing_algo: algo,
-        };
+        // Generate and store both ECDSA and ED25519 signatures
+        for algo in ["ecdsa", "ed25519"] {
+            let (signature_hex, signing_address) = match algo {
+                "ed25519" => {
+                    let signature_bytes = self.ed25519_signing_key.sign(signature_text.as_bytes());
+                    let sig_hex = hex::encode(signature_bytes.to_bytes());
+                    let addr = self.get_signing_address_hex("ed25519");
+                    Ok((sig_hex, addr))
+                }
+                "ecdsa" => {
+                    // Sign using ECDSA with recovery ID
+                    // Use Ethereum signed message format
+                    let message_bytes = signature_text.as_bytes();
+                    let prefix = format!("\x19Ethereum Signed Message:\n{}", message_bytes.len());
+                    let prefix_bytes = prefix.as_bytes();
+
+                    // Concatenate prefix + message
+                    let mut prefixed_message =
+                        Vec::with_capacity(prefix_bytes.len() + message_bytes.len());
+                    prefixed_message.extend_from_slice(prefix_bytes);
+                    prefixed_message.extend_from_slice(message_bytes);
+
+                    // Hash with Keccak256 (manually hash the prefixed message)
+                    let mut hasher = Keccak256::new();
+                    hasher.update(&prefixed_message);
+                    let message_hash = hasher.finalize();
+
+                    // Use sign_prehash_recoverable with the pre-hashed message
+                    let (signature, recid): (EcdsaSignature, RecoveryId) = self
+                        .ecdsa_signing_key
+                        .sign_prehash_recoverable(&message_hash)
+                        .map_err(|e| {
+                            tracing::error!("Failed to create recoverable ECDSA signature: {}", e);
+                            AttestationError::InternalError(format!(
+                                "Failed to create recoverable ECDSA signature: {e}"
+                            ))
+                        })?;
+
+                    // Convert signature to bytes and append recovery ID
+                    // Convert k256 RecoveryId (0-3) to Ethereum v format (27-28)
+                    // Ethereum v = 27 + (recovery_id & 1) where bit 0 is the y-coordinate parity
+                    let recovery_byte = recid.to_byte();
+                    let ethereum_v = 27u8 + (recovery_byte & 1);
+
+                    // This creates a 65-byte signature (64 bytes r||s + 1 byte Ethereum v)
+                    let mut signature_bytes = signature.to_bytes().to_vec();
+                    signature_bytes.push(ethereum_v);
+                    let sig_hex = hex::encode(signature_bytes);
+
+                    let addr = self.get_signing_address_hex("ecdsa");
+                    Ok((format!("0x{sig_hex}"), addr))
```

Comment:
Why prefix ecdsa with 0x. When we don't for ed25519?

Reply:
This is to be consistent with the behavior returned by the model